src/client/pages/OfferNew/Introduction/Sidebar/DetailsModal/utils.ts

```diff
@@ -627,80 +627,133 @@ export const getInitialInputValues = (
   }
 }
 
-export const getInitialQuoteDetailsInputValues = (offerQuote: OfferQuote) =>
-  match<string, EditQuoteInput>([
-    [
-      'SwedishHouseQuoteDetails',
-      () =>
-        getInitialSwedishHouseValues(
-          offerQuote.id,
-          offerQuote.quoteDetails as SwedishHouseQuoteDetails,
-        ),
-    ],
-    [
-      'SwedishApartmentQuoteDetails',
-      () =>
-        getInitialSwedishApartmentValues(
-          offerQuote.id,
-          offerQuote.quoteDetails as SwedishApartmentQuoteDetails,
-        ),
-    ],
-    [
-      'NorwegianHomeContentsDetails',
-      () =>
-        getInitialNorwegianHomeContentValues(
-          offerQuote.id,
-          (offerQuote.quoteDetails as QueryQuoteDetails) as QueryQuoteDetails & {
-            __typename: 'NorwegianHomeContentsDetails'
-          },
-        ),
-    ],
-    [
-      'NorwegianTravelDetails',
-      () =>
-        getInitialNorwegianTravelValues(
-          offerQuote.id,
-          offerQuote.quoteDetails as NorwegianTravelDetails,
-        ),
-    ],
-    [
-      'DanishHomeContentsDetails',
-      () =>
-        getInitialDanishHomeContentValues(
-          offerQuote.id,
-          (offerQuote.quoteDetails as QueryQuoteDetails) as {
-            __typename: 'DanishHomeContentsDetails'
-          } & QueryQuoteDetails,
-        ),
-    ],
-  ])(offerQuote.quoteDetails.__typename as string) ??
-  (() => {
-    throw new window.Error(
-      `Expected quote details to be a valid type but was "${offerQuote.quoteDetails.__typename}"`,
-    )
-  })()
+export const getInitialQuoteDetailsInputValues = (offerQuote: OfferQuote) => {
+  const { id: quoteId, quoteDetails } = offerQuote
+
+  switch (quoteDetails.__typename) {
+    case 'SwedishHouseQuoteDetails':
+      return getInitialSwedishHouseValues(
+        quoteId,
+        quoteDetails as SwedishHouseQuoteDetails,
```

Comment:
So some of these type assertions could be removed so I fixed that 👍  I didn't really look into that before, just changed the structure to a `switch` statement, but typescript _can_ assert this - the issue is with `NorwegianHomeContentsDetails` and `DanishHomeContentsDetails` since they due to reasons have special typing 🙃